src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,198 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V>[] currentTable = table;
+        int hash = hash(key);
+        if (currentTable == null || currentTable.length == 0) {
+            currentTable = resize();
+        }
+        if (currentTable[hash % currentTable.length] == null) {
+            currentTable[hash % currentTable.length] = new Node<>(hash, key, value, null);
+        } else {
+            Node<K, V> currentNode = currentTable[hash % currentTable.length];
+            if ((currentNode.key == null) && (key == null)) { // если обе нулл
```

Comment:
```suggestion
            if ((currentNode.key == null) && (key == null)) { 
```
comment is not needed